fenix/app/src/main/java/org/mozilla/fenix/components/accounts/FenixFxAEntryPoint.kt

```diff
@@ -0,0 +1,111 @@
+
+package org.mozilla.fenix.components.accounts
+
+import android.os.Parcel
+import android.os.Parcelable
+import mozilla.components.concept.sync.FxAEntryPoint
+
+/**
+ * Fenix implementation of [FxAEntryPoint].
+ */
+enum class FenixFxAEntryPoint(override val entryName: String) : FxAEntryPoint, Parcelable {
+    /**
+     * New user onboarding, the user accessed the sign in through new user onboarding
+     */
+    NewUserOnboarding("newuser-onboarding"),
+
+    /**
+     * Manual sign in from the onboarding menu
+     */
+    OnboardingManualSignIn("onboarding-manual-sign-in"),
+
+    /**
+     * User used a deep link to get to firefox accounts authentication
+     */
+    DeepLink("deep-link"),
+
+    /**
+     * Authenticating from the browser's toolbar
+     */
+    BrowserToolbar("browser-toolbar"),
+
+    /**
+     * Authenticating from the home menu (the hamburger menu)
+     */
+    HomeMenu("home-menu"),
+
+    /**
+     * Authenticating in the bookmark view, when getting attempting to get synced
+     * bookmarks
+     */
+    BookmarkView("bookmark-view"),
+
+    /**
+     * Authenticating from the homepage onboarding dialog
+     */
+    HomeOnboardingDialog("home-onboarding-dialog"),
+
+    /**
+     * Authenticating from the settings menu
+     */
+    SettingsMenu("settings-menu"),
+
+    /**
+     * Authenticating from the autofill settings to enable synced
+     * credit cards/addresses
+     */
+    AutofillSetting("autofill-setting"),
+
+    /**
+     * Authenticating from the saved logins menu to enable synced
+     * logins
+     */
+    SavedLogins("saved-logins"),
+
+    /**
+     * Authenticating from the Share menu to enable send tab
+     */
+    ShareMenu("share-menu"),
+
+    /**
+     * Authenticating as a navigation interaction
+     */
+    NavigationInteraction("navigation-interaction"),
+
+    /**
+     * Authenticating from the synced tabs menu to enable synced tabs
+     */
+    SyncedTabsMenu("synced-tabs-menu"),
+
+    /**
+     * When serializing the value after navigating, the result is a nullable value. We have this
+     * "unknown" as a default value in the odd chance that we receive an [entryName] is not part of this enum.
+     */
+    Unknown("unknown"),
+    ;
+
+    override fun writeToParcel(parcel: Parcel, flags: Int) {
+        parcel.writeString(entryName)
+    }
+
+    override fun describeContents(): Int {
+        return 0
+    }
+
+    /**
+     * Override implementation of the [Parcelable.Creator].
+     */
+    companion object CREATOR : Parcelable.Creator<FenixFxAEntryPoint> {
+        override fun createFromParcel(parcel: Parcel): FenixFxAEntryPoint {
+            val parcelEntryName = parcel.readString() ?: Unknown
+            return FenixFxAEntryPoint.values().first { it.entryName == parcelEntryName }
+        }
+
+        override fun newArray(size: Int): Array<FenixFxAEntryPoint?> {
+            return arrayOfNulls(size)
+        }
+    }
```

Comment:
@tarikeshaq @MatthewTighe this is ready for a final review, but before landing, I'd like to draw your attention to this addition I had to make.

The nav graph needs the argument parameter to be `Parcelable` so this part was partially generated. The `createFromParcel` function introduced an `Unknown` entrypoint for compile-time safety.

Reply:
Would this boilerplate be avoidable with the `@Parcelize` [annotation](https://developer.android.com/kotlin/parcelize)?

Reply:
Yeah, I tried that first. It didn't work because it's an enum so it couldn't figure out how to return the default value for `createFromParcel`. It kept trying to return a `FenixFxAEntryPoint` and not a _value type_ of `FenixFxAEntryPoint`, so the override was needed.

Then the next problem: `parcel.readString()` was returning a nullable string which can occur from the OS crapping out during navigation too on some rare cases. Crashing because the entry point couldn't be serialized didn't see like a good enough reason.

Reply:
Ah gosh, thank you for wrestling with this, I might have come across a form of this when doing the first version of this but I can't recall how it worked.

+1 from me - and the comment on the `Unknown` seems clear enough that consumers shouldn't set it manually

Reply:
I added some more comments in this area to explain the reasoning since it isn't obvious why we had to write some of this boilerplate.

Thanks for bringing it up!